cadc-storage-adapter/src/test/java/org/opencadc/inventory/storage/TestStorageAdapter.java

```diff
@@ -0,0 +1,214 @@
+            byte[] md5 = md.digest(data);
+            String md5hex = HexUtil.toHex(md5);
+            contentChecksum = URI.create("md5:" + md5hex);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    static Mode mode = Mode.NORMAL;
+    
+    public enum Mode { NORMAL, ERROR_ON_GET_0, ERROR_ON_GET_1, ERROR_ON_GET_2 };
+    static int BUF_SIZE = 6;
+    
+    public TestStorageAdapter() {
+    }
+
+    @Override
+    public void get(StorageLocation storageLocation, InputStreamWrapper wrapper)
+            throws ResourceNotFoundException, IOException, TransientException {
+        InputStream in = null;
+        if (mode.equals(Mode.ERROR_ON_GET_0)) {
+            in = new ErrorInputStream(data, 0);
+        } else if (mode.equals(Mode.ERROR_ON_GET_1)) {
+            in = new ErrorInputStream(data, 1);
+        } else if (mode.equals(Mode.ERROR_ON_GET_2)) {
+            in = new ErrorInputStream(data, 2);
+        } else {
+            in = new ByteArrayInputStream(data);
+        }
+        wrapper.read(in);
+    }
+
+    @Override
+    public void get(StorageLocation storageLocation, InputStreamWrapper wrapper, Set<String> cutouts)
+            throws ResourceNotFoundException, IOException, TransientException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public StorageMetadata put(Artifact artifact, OutputStreamWrapper wrapper)
+            throws StreamCorruptedException, IOException, TransientException {
+        
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        wrapper.write(out);
+        byte[] newData = out.toByteArray();
+        log.info("received data: " + new String(newData));
+        
+        URI newContentChecksum = null;
+        try {
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            byte[] md5 = md.digest(newData);
+            String md5hex = HexUtil.toHex(md5);
+            newContentChecksum = URI.create("md5:" + md5hex);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        long newContentLength = newData.length;
+        if (!newContentChecksum.equals(contentChecksum)) {
+            throw new StreamCorruptedException("checksum: " +
+                newContentChecksum + " does not equal " + contentChecksum);
+        }
+        if (newContentLength != contentLength) {
+            throw new StreamCorruptedException("length: " +
+                newContentLength + " does not equal " + contentLength);
+        }
+        
+        StorageLocation storageLocation = new StorageLocation(storageID);
+        StorageMetadata storageMetadata = new StorageMetadata(storageLocation, contentChecksum, contentLength);
+        storageMetadata.artifactURI = artifact.getURI();
+        return storageMetadata;
+    }
+
+    @Override
+    public void delete(StorageLocation storageLocation) throws ResourceNotFoundException, IOException, TransientException {
```

Comment:
Is these methods are not implemented they should throw an UnsupprotedOperationException